anesthetic/plotting/_matplotlib/core.py

```diff
@@ -0,0 +1,101 @@
+from pandas.plotting._matplotlib.core import (ScatterPlot as _ScatterPlot,
+                                              HexBinPlot as _HexBinPlot,
+                                              LinePlot as _LinePlot,
+                                              BarPlot as _BarPlot,
+                                              BarhPlot as _BarhPlot,
+                                              AreaPlot as _AreaPlot,
+                                              PiePlot as _PiePlot,
+                                              MPLPlot)
+from anesthetic.weighted_pandas import _WeightedObject
+import numpy as np
+
+
+def _get_weights(kwargs, data):
+    if isinstance(data, _WeightedObject):
+        kwargs['weights'] = data.weights
+
+
+class _WeightedMPLPlot(MPLPlot):
+    def __init__(self, data, *args, **kwargs):
+        _get_weights(kwargs, data)
+        super().__init__(data, *args, **kwargs)
+
+    def _get_index_name(self):
+        if isinstance(self.data, _WeightedObject):
+            return np.squeeze(self.data.index.droplevel('weights').names)
+        else:
+            return super()._get_index_name()
+
+    def _get_xticks(self, convert_period: bool = False):
+        if isinstance(self.data, _WeightedObject):
+            return self.data.index.droplevel('weights')._mpl_repr()
+        else:
+            return super()._get_xticks(convert_period)
```

Comment:
This does seem to fix the issue of having unwelcome `weights` on the x-axis. However, this also seems to remove the x-ticklabels when we actually want them. Try:
```python 
ns = NestedSamples(root="anesthetic/tests/example_data/pc")
wdf = WeightedDataFrame(ns)
wdf.plot.hexbin('x0', 'x1')
```

![image](https://user-images.githubusercontent.com/18258042/183755433-ec0e5296-813d-46d8-95e9-bd79d1b523aa.png)


The x-ticklabels are gone, even though we would like to have the labels from `'x0'`.

Reply:
Actually, no, this seems to be the (ugly) default behaviour of pandas, so nevermind...